Nested groups

scottsykora opened the pull request:
This merges all the changes we've added to support groups of arrays as a way to return the parsed cgpaths. This allows our Zinnia parser to separate the svg into layers with different colors and use the groups as a way to find our own tags.

This is not really meant to merge back into the upstream repo ever. I don't think this is the way they would build this more general functionality if they did want it. It's more of a hack to get ours working with their general codebase.

scottsykora committed:
Adds code to optionally parst svgs into nested groups of CGPathRefs

scottsykora committed:
Updates code to support grouped arrays as a way to parse cgpaths

scottsykora committed:
Fixes some bugs in the grouping code (brackets around if statements)

premedrchang committed:
fix issue with crashing on grouped svgs AND run with compiler optimization.
issue was with a boolean not specifically set to FALSE.
added a test with a grouped svg and set the project to default to smallest/fastest optimization on the debug config

scottsykora committed:
Merge branch 'master' into nested_groups

scottsykora commented:
FYI, I think it's ok that some of the checks above fail. I think those are all part of the parent repo, the main thing is that the spm and cocoa pods checks pass and the tests inside the spm pass locally.

designatednerd commented:
If we're planning on maintaining this as a separate repo and not trying to get this merged upstream, does it seem like maybe we should disable the upstream checks?

scottsykora commented:
I'm not sure... I still want to be able to sync and merge changes from the upstream back into this fork... I assume disabling upstream checks would break that?

designatednerd commented:
I would imagine if you're merging upstream changes, unless they're actually on the code that runs the tests, they shouldn't cause conflicts. But I'm fine leaving it as is if that's what you'd prefer.

scottsykora commented:
@designatednerd so there's a way to just turn off the upstream checks but leave it linked as a fork? How do I do that in GitHub? I have no idea and google didn't seem to reveal anything obvious.

designatednerd commented:
I think you should be able to comment out the failing bits here: https://github.com/pixiteapps/PocketSVG/blob/master/.github/workflows/primary.yml - I believe if you do that, only the other checks will run, and that won't affect any upstream repos since the changes are only in this repo.

scottsykora committed:
Commment out a test

scottsykora committed:
Remove Dynamic Target

Removing the dynamic target type allows this to compile correctly when compiling for catalyst.

scottsykora committed:
Merge pull request #2 from pixiteapps/ss/make_static

Remove Dynamic Target

scottsykora committed:
Revert to previous parsing method

This reverts to the previous method of parsing numbers, it seems like the pr that updated this broke it for a lot of our vectors.
